src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlySpanND.cs

```diff
@@ -0,0 +1,468 @@
+
+        /// <summary>
+        /// Creates a new read-only span over the entirety of the target array.
+        /// </summary>
+        /// <param name="array">The target array.</param>
+        /// <param name="lengths">The <see cref="System.ReadOnlySpan{T}"/> with the lengths of each dimension.</param>
+        /// <remarks>Returns default when <paramref name="array"/> is null.</remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public ReadOnlySpanND(T[]? array, ReadOnlySpan<nint> lengths)
+        {
+            if (array == null)
+            {
+                this = default;
+                return; // returns default
+            }
+
+            _linearLength = SpanNDHelpers.CalculateTotalLength(lengths);
+            if (_linearLength != array.Length)
+                ThrowHelper.ThrowArgument_LengthsMustEqualArrayLength();
+
+            _reference = ref MemoryMarshal.GetArrayDataReference(array);
+            _linearLength = array.Length;
+            _lengths = lengths.ToArray().AsSpan();
+            _strides = SpanNDHelpers.CalculateStrides(Rank, lengths);
+        }
+
+        /// <summary>
+        /// Creates a new read-only span over the portion of the target array beginning
+        /// at 'start' index and ending at 'end' index (exclusive).
+        /// </summary>
+        /// <param name="array">The target array.</param>
+        /// <param name="start">The index at which to begin the read-only span.</param>
+        /// <param name="lengths">The <see cref="System.ReadOnlySpan{T}"/> with the lengths of each dimension.</param>
+        /// <remarks>Returns default when <paramref name="array"/> is null.</remarks>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when the specified <paramref name="start"/> or end index is not in the range (&lt;0 or &gt;Length).
+        /// </exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public ReadOnlySpanND(T[]? array, nint start, ReadOnlySpan<nint> lengths)
+        {
+            _linearLength = SpanNDHelpers.CalculateTotalLength(lengths);
+            if (array == null)
+            {
+                if (start != 0 || _linearLength != 0)
+                    ThrowHelper.ThrowArgumentOutOfRangeException();
+                this = default;
+                return; // returns default
+            }
+
+#if TARGET_64BIT
+            // See comment in SpanND<T>.Slice for how this works.
+            if ((ulong)(uint)start + (ulong)(uint)length > (ulong)(uint)array.Length)
+                ThrowHelper.ThrowArgumentOutOfRangeException();
+#else
+            if ((uint)start > (uint)array.Length || (uint)_linearLength > (uint)(array.Length - start))
+                ThrowHelper.ThrowArgumentOutOfRangeException();
+#endif
+            if (_linearLength != SpanNDHelpers.CalculateTotalLength(lengths))
+                ThrowHelper.ThrowArgument_LengthsMustEqualArrayLength();
```

Comment:
This is redoing the computation already done above

Reply:
Fixed.